### ISSUE TITLE
Add support for ARM Linux and ARM64 Windows

### ISSUE THREAD

**Issue by hmsjy2017** [typora/typora-issues#4282]
It is no exaggeration to say that Typora is the best Markdown editor I have ever used. I have searched for a long time, but I have never found a Markdown editor that is better than Typora. The lack of support for ARM has always troubled me and many people.
At present, Typora can't run on many popular ARM computers such as Raspberry Pi at all, which brings us great inconvenience. Because there is no native ARM support, it currently runs very slowly on Windows 10 ARM64, but many people cannot do without it.
I think it's high time to add support for ARM. Nowadays, more and more software have joined the support for ARM, including the famous VS Code and the popular GIMP. This shows that ARM's prospects are very bright. I believe that after adding support for ARM, more people will use Typora to make Typora more powerful.
I sincerely hope you can adopt my suggestion. Thank you!😄

**Comment by abnerlee** [comment]
#2524

**Comment by michalsc** [comment]
I am aware that you have refused such requests previously, but I think it is about the right time to reconsider it. I am not talking about AArch32, which is full of weak processors and incompatible instruction sets. What I mean is AArch64 world - well standarized ISA with mandatory FPU onboard. The 64bit ARM CPUs are powerful enough to run Typora. Well, actually you have just ported Typora to first AArch64 machine - Apple M1 CPU.

If you can support M1, why not supporting AArch64 Linux or AArch64 windows? If RaspberryPi or PinebookPro is strong enough to run Visual Studio Code based on Electron, it would just run Typora without any issue.

**Comment by abnerlee** [comment]
I currently don't have any arm devices for developing and testing. But now I think currently arm devices may be powerful enough to run typora, so we will reconsider it.

**Comment by michalsc** [comment]
> I currently don't have any arm devices for developing and testing. But now I think currently arm devices may be powerful enough to run typora, so we will reconsider it.

If you need any help on arm/aarch64 side with developing or test let me know. I will be happy to help on that.

**Comment by Herby007** [comment]
Raspberry Pi 400 runs very well with 64-Bit systems like Manjaro. More and more people using this machine (or Raspberry Pi 4 with 4-GB or 8-GB) as a real second cheap PC. Would be absolutly great to see Typora there!

**Comment by plumlis** [comment]
It seems typora support Mac OS M1 chip now.
https://support.typora.io/What's-New-0.9.98/
So shall we can hope for arm64 Win or Linux support?

**Comment by hmsjy2017** [comment]
Typora is based on Electron, so it will not be too difficult to support arm64 Linux and Windows. The key is whether Typora is willing to support the arm64 architecture of other platforms.

**Comment by abnerlee** [comment]
I don't have arm devices for developing and testing. But it should be possible after I get my M1X / M2 macbook

**Comment by hmsjy2017** [comment]
@abnerlee If you do plan to support Linux/Windows arm64, I am willing to  provide help.

**Comment by abnerlee** [comment]
> I would like to ask, without a Mac with M1 chip, how did you build Typora that supports the M1 chip?

We build an universal app on Intel Mac, but windows / Linux does not offer such option.

---

I also tried with a browered arm64 PC, but not working with centOS -> https://github.com/electron/electron/issues/25387
It seems takes time for trouble shooting

---

And we may not target for Raspberry Pi, but real ARM PC. Typora on Linux/Windows is based on Electron / Chromium, so I guess the performance may be not OK for everyday usage.

**Comment by hmsjy2017** [comment]
@abnerlee Why not use Debian or Ubuntu? I don't think anyone will use CentOS as a desktop system.
****
The Raspberry Pi is indeed an ARM PC, because it can do almost everything other ARM PCs can do. As for your concerns about the performance on the Raspberry Pi, I don't think you need to worry about it.
 At present, Electron-based VS Code already supports Raspberry Pi (including armhf and arm64), and the experience makes me feel very satisfied (the startup time is only 8 seconds, and the startup time of Chromium is 5 seconds). I do not agree that the performance of the Raspberry Pi is not enough to run Typora.

**Comment by abnerlee** [comment]
> Why not use Debian or Ubuntu? I don't think anyone will use CentOS as a desktop system.

But there's only Ubuntu Server for arm64 (https://ubuntu.com/download/server/arm) and Ubuntu for Raspberry Pi (https://ubuntu.com/raspberry-pi), no normal desktop version of Ubuntu Arm64. Quit unsure how users use arm64 PCs...

**Comment by plumlis** [comment]
Using arm64 Linux for desktop is quite eazy.
Just install Ubuntu-desktop package on arm64 Server Edition, and I works quite well.
I had a Pinebook Pro which running Ubuntu Arm64 20.04 and It's pretty usable, But It lacks my favourate Typora.

Also some Arm64 Chromebook with Linux sub system will need arm64 version.

**Comment by michalsc** [comment]
There is e.g. 64-bit ARM manjaro linux available both for PineBook Pro and for Raspberry:

https://manjaro.org/downloads/arm/raspberry-pi-4/arm8-raspberry-pi-4-xfce/

**Comment by lu-min** [comment]
I am using an Arm-based Chromebook and tried to install Typora in its Linux subsystem (I did with an x86 CPU Chromebook before) but found there is no arm64 version yet. It will be great and helpful if an arm version is available.

**Comment by triantares** [comment]
> > Why not use Debian or Ubuntu? I don't think anyone will use CentOS as a desktop system.
> 
> But there's only Ubuntu Server for arm64 (https://ubuntu.com/download/server/arm) and Ubuntu for Raspberry Pi (https://ubuntu.com/raspberry-pi), no normal desktop version of Ubuntu Arm64. Quite unsure how users use arm64 PCs...

Well there are quite a few distros that run very well on the Pinebook Pro, including Manjaro and Ubuntu as well as Debian so that in a sense is a sensible arm64 machine to test on IMO.

**Comment by abnerlee** [comment]
We will release them along with v1.0 soon.

You can now test a pre build version from https://download.typora.io/linux/typora_0.11.18_arm64.deb

**Comment by hmsjy2017** [comment]
@abnerlee Thanks !

**Comment by abnerlee** [comment]
supported now

**Comment by youngd24** [comment]
> I don't have arm devices for developing and testing. But it should be possible after I get my M1X / M2 macbook

A Pi 4 is all that is needed for testing. I'm running a Pi 4 as my IoT lab workstation (dual-head) and it runs everything I need to do that development work. I'd prefer to buy a copy of Typora for it rather than continue to use M$ vscode. The Pi as of the 4 (and the 400) is now a viable small computer. The issue lately is actually getting your hands on one.

I'll try that pre-build version and see how it goes but wanted to toss in how it's well worth the time to support linux/arm/Pi now.

**Comment by hmsjy2017** [comment]
@youngd24 It runs very well on the Raspberry Pi, just like it does on Windows.

**Comment by youngd24** [comment]
Thanks, I’ll give it a shot on the 64-bit card I have. Not everything works under arm64, mostly containers, so I’m currently flipping back and forth between.

**Comment by GertVanKruiningen** [comment]
**Wow**, just tried, not thinking anything would work on a PinePhone Pro.
But it did install and show a user interface, and opened settings....
But that's where it stopped working
No idea if there's still development happening, or if it should even work on an PPP...
Let me know

**Comment by caseydiers** [comment]
I have been using for years on windows and linux, and I'm glad I checked this morning to see if typora supports arm and found this thread.

I installed the latest version on my PineBook Pro (arm processor on Manjaro) and it opened right away.  On the first run, I was able to type but upon trying to open a menu it crashed right away.  I opened it again, and everything has been working as expected since.  Thank you so much for supporting this.  PineBook Pro is great hardware for writing, and Typora is greatest software for writing.

As a side note, I tried installing from the AUR and it said that aarch64 wasn't supported.  So I downloaded the tar from the website and it worked right away.

I'll keep using an post any issues I find along the way.
